sdk/cosmosdb/cosmos/src/request/request.ts

```diff
@@ -127,7 +127,11 @@ export async function getHeaders({
     headers[Constants.HttpHeaders.ConsistencyLevel] = options.consistencyLevel;
   }
 
-  if (options.maxIntegratedCacheStalenessInMs && resourceType === ResourceType.item) {
+  if (
+    options.maxIntegratedCacheStalenessInMs &&
+    options.maxIntegratedCacheStalenessInMs > 0 &&
```

Comment:
Developer might not realize silent ignore. Better alternative is to error out with equivalent of ArgumentException while setting in-correct value (i.e. -ve value).

Reply:
I thought about it, but I simply exclude the header from the request (similar to the backend logic).  This is the equivalent to setting the default. I can log the message to the console for the developer. JavaScript does not have an ArgumentException or  InvalidArgumentException error.

Reply:
Once a features GA's any future behaviors changes will be treated as breaking changes.
What's the JS way of doing argument validation?

Reply:
We typically do very little argument validation (for models at least) and let the server handle sending back appropriate errors. For cases like this where you're controlling local behavior I think throwing a RangeError: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError is fine

Reply:
@xirzec is the suggestion to pass through the values and let service handle the validation? 

That's one reasonable option where client is pure pass-through (that's the current status). 
@jay-most if validation or any is needed on client then better fail gracefully with RangeError.

Reply:
> @xirzec is the suggestion to pass through the values and let service handle the validation?

Whenever possible that is our preference, since sometimes the service expands its handling and it's nice to not force folks to update their SDKs because we had to remove/update logic in the client.